include/mg_procedure.h

```diff
@@ -1405,36 +1405,59 @@ int mgp_must_abort(struct mgp_graph *graph);
 
 /// @}
 
-/// @name Kafka message API
-/// Currently the API below is for kafka only but in the future
-/// mgp_message and mgp_messages might be generic to support
-/// other streaming systems.
+/// @name Stream Source message API
+/// API for accessing specific data contained in a mgp_message
+/// used for defining transformation procedures.
+/// Not all methods are available for all stream sources
+/// so make sure that your transformation procedure can be used
+/// for a specific source, i.e. only valid methods are used.
 ///@{
 
-/// A single Kafka message
+/// A single Stream source message
 struct mgp_message;
 
-/// A list of Kafka messages
+/// A list of Stream source messages
 struct mgp_messages;
 
+/// Get the type of the stream source that produced the message.
+/// Valid values:
+///   - "kafka"
+///   - "pulsar"
+enum mgp_error mgp_message_source_type(struct mgp_message *message, const char **result);
```

Comment:
Why not an enumeration value?

Reply:
I wasn't sure how to handle it on the python side. This seemed like the easiest way, we don't rely on the underlying values and the user gets the same output as the SHOW STREAMS query.